sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -166,18 +193,72 @@ def create_propagator(injectors, extractors)
         end
       end
 
-      def default_injectors
-        [
-          OpenTelemetry::Trace::Propagation::TraceContext.text_map_injector,
-          OpenTelemetry::Baggage::Propagation.text_map_injector
-        ]
+      def b3_single_propagator
+        if defined?(OpenTelemetry::Propagator::B3::Single)
+          [OpenTelemetry::Propagator::B3::Single.text_map_injector, OpenTelemetry::Propagator::B3::Single.text_map_extractor]
+        else
+          unavailable_external_propagator 'b3'
+        end
+      end
+
+      def b3_multi_propagator
+        if defined?(OpenTelemetry::Propagator::B3::Multi)
+          [OpenTelemetry::Propagator::B3::Multi.text_map_injector, OpenTelemetry::Propagator::B3::Multi.text_map_extractor]
+        else
+          unavailable_external_propagator 'b3multi', 'b3'
+        end
+      end
+
+      def jaeger_propagator
+        if defined?(OpenTelemetry::Propagator::Jaeger)
+          [OpenTelemetry::Propagator::Jaeger.text_map_injector, OpenTelemetry::Propagator::Jaeger.text_map_extractor]
+        else
+          unavailable_external_propagator 'jaeger'
+        end
+      end
+
+      def xray_propagator
+        if defined?(OpenTelemetry::Propagator::XRay)
+          [OpenTelemetry::Propagator::XRay.text_map_injector, OpenTelemetry::Propagator::XRay.text_map_extractor]
+        else
+          unavailable_external_propagator 'xray'
+        end
+      end
+
+      def ot_trace_propagator
+        if defined?(OpenTelemetry::Propagator::OTTrace)
+          [OpenTelemetry::Propagator::OTTrace.text_map_injector, OpenTelemetry::Propagator::OTTrace.text_map_extractor]
+        else
+          unavailable_external_propagator 'ottrace'
+        end
+      end
+
+      def unavailable_external_propagator(name, gem_suffix = name)
+        OpenTelemetry.logger.warn "The #{name} propagator cannot be configured - please add opentelemetry-propagator-#{gem_suffix} to your Gemfile"
+        [nil, nil]
+      end
+
+      def otlp_exporter
```

Comment:
Nitpick: I have concerns with the "clutter" associated with adding all these `if defined?(` checking methods to the configurator class.  I would like to see these extracted to some external module, or even common. Maybe something like `OpenTelemetry::Common::ExporterFromEnv.otlp_exporter`, the helper could just as easily exist in the SDK.  I don't have too strong of a preference where.

Reply:
I've taken a different approach, based on your suggestions in Slack. 🙏 LMK if this is cleaner or if you still want it extracted.